frontend/benefit/applicant/src/components/applications/alteration/AlterationPage.tsx

```diff
@@ -0,0 +1,198 @@
+import { AxiosError } from 'axios';
+import {
+  $BackButtonContainer,
+  $MainHeaderItem,
+  $PageHeader,
+  $PageHeading,
+} from 'benefit/applicant/components/applications/alteration/AlterationPage.sc';
+import useAlterationPage from 'benefit/applicant/components/applications/alteration/useAlterationPage';
+import {
+  $HeaderItem,
+  $HeaderRightColumnItem,
+  $PageHeadingApplicant,
+  $PageSubHeading,
+  $SpinnerContainer,
+} from 'benefit/applicant/components/applications/Applications.sc';
+import AlterationForm from 'benefit/applicant/components/applications/forms/application/alteration/AlterationForm';
+import ErrorPage from 'benefit/applicant/components/errorPage/ErrorPage';
+import { ROUTES } from 'benefit/applicant/constants';
+import {
+  ALTERATION_STATE,
+  ALTERATION_TYPE,
+  APPLICATION_STATUSES,
+} from 'benefit-shared/constants';
+import { ApplicationAlterationData } from 'benefit-shared/types/application';
+import camelcaseKeys from 'camelcase-keys';
+import { Button, IconArrowLeft, LoadingSpinner } from 'hds-react';
+import kebabCase from 'lodash/kebabCase';
+import { useRouter } from 'next/router';
+import React from 'react';
+import Container from 'shared/components/container/Container';
+import hdsToast from 'shared/components/toast/Toast';
+import { convertToUIDateAndTimeFormat } from 'shared/utils/date.utils';
+import { useTheme } from 'styled-components';
+
+const AlterationPage = (): JSX.Element => {
+  const { t, id, application, isError, isLoading } = useAlterationPage();
+
+  const router = useRouter();
+  const theme = useTheme();
+
+  if (isLoading) {
+    return (
+      <$SpinnerContainer>
+        <LoadingSpinner />
+      </$SpinnerContainer>
+    );
+  }
+
+  if (isError || !id) {
+    return (
+      <ErrorPage
+        title={t('common:errorPage.title')}
+        message={t('common:errorPage.message')}
+        showActions={{ linkToRoot: true, linkToLogout: true }}
+      />
+    );
+  }
+
+  const returnToApplication = (): void =>
+    void router.push(`${ROUTES.APPLICATION_FORM}?id=${id}`);
+
+  const onSuccess = async (
+    response: ApplicationAlterationData
+  ): Promise<void> => {
+    returnToApplication();
+
+    const textKey =
+      response.alteration_type === ALTERATION_TYPE.TERMINATION
+        ? 'common:applications.alteration.successToast.bodyTermination'
+        : 'common:applications.alteration.successToast.bodySuspension';
+
+    hdsToast({
+      autoDismissTime: 0,
+      type: 'success',
+      labelText: t('common:applications.alteration.successToast.title'),
+      text: t(textKey, {
+        id: application.ahjoCaseId,
+      }),
+    });
+  };
+
+  const onError = (error: AxiosError<unknown>): void => {
```

Comment:
Just a note. I find myself writing new error handlers every now and then. We have a ticket to generalize all errors and I hope to see it being done in the future.